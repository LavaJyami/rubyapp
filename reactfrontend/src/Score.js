import React, { Component } from 'react';

class Score extends Component {
  constructor(props){
    super(props);
    this.state = {
      score: 0
    }
  }
  showScore(){
    var length = this.props.words.length;
    var totalScore = 0;
    if(length > 0){
      for(let i=0;i<length;i++){
        totalScore = totalScore + this.props.words[i].length;
      }
      return totalScore;
    }
    else
    return 0;
  }
  render(){
    return(
      <div id="score">
        Score: {this.showScore()}
      </div>
    );
  }
}

export default Score;
