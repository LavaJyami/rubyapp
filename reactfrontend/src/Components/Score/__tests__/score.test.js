import React from 'react';
import ReactDOM from 'react-dom';
import Score from '../Score';

import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

afterEach(cleanup);

var wordsArr = ["apple", "oranges", "kiwi", "mango"];

it("Score renders without crashing",() => {
  const div = document.createElement("div");
  ReactDOM.render(<Score words={wordsArr} />, div);
});


it("Score is calculated correctly",() => {
  const {getByTestId} = render(<Score words = {wordsArr}/>);
  expect (getByTestId('score')).toHaveTextContent("21");
});
