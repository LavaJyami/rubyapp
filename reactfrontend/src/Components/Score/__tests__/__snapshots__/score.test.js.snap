// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="score"
  id="score"
>
  Score: 
  21
</div>
`;
